pkg/limayaml/limayaml.go

```diff
@@ -31,6 +31,7 @@ type LimaYAML struct {
 	Env          map[string]string `yaml:"env,omitempty" json:"env,omitempty"`
 	DNS          []net.IP          `yaml:"dns,omitempty" json:"dns,omitempty"`
 	HostResolver HostResolver      `yaml:"hostResolver,omitempty" json:"hostResolver,omitempty"`
+	UserNet      UserNet           `yaml:"userNet,omitempty" json:"userNet,omitempty"`
```

Comment:
Shouldn't this be in `Networks`?

Reply:
Didn't add under networks because as of now Networks are additional config for us, 
but this will replace the behaviour of base network itself. So went ahead with a separate one.

Reply:
I'd expect the default networking to be eventually configurable in `networks.yaml` 
```yaml
networks:
  default:
    mode: user-v2
    gateway: 192.168.5.1
    dhcpEnd: 192.168.5.254
    netmask: 255.255.255.0
  shared:
    mode: shared
    gateway: 192.168.105.1
    dhcpEnd: 192.168.105.254
    netmask: 255.255.255.0
```

When it is out of experimental, the `default` network will be automatically prepended to the network list to replace the legacy slirp network

Reply:
Ah it makes sense. I made usernet as object mainly to provide this configuration later. But missed the part of configuring via networks.yaml 

Will move it under networks

Reply:
Done the changes. 
Added new property under Networks called UserNet to point to mode value networks.yml. Kept the mode as user-v2 for now.

Based on below discussion we can finalise a name and change it here as well accordingly

Reply:
We could reuse Network.Lima property also. But we will end up needing to resolve networks as well. To identify if the Lima property is for user-v2 or socker_vmnet. 

Added UserNet to make this simple and keep this separate. If you think this will cause confusion happy to change code to reuse Lima property

Reply:
> But we will end up needing to resolve networks as well. To identify if the Lima property is for user-v2 or socker_vmnet.

Yes, is this problematic?

Reply:
`Yes, is this problematic?`
I thought it would be, but it is not actually. Done the changes with Lima property itself.